alert-common/src/main/java/com/synopsys/integration/alert/common/enumeration/AuthenticationPriority.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * alert-common
+ *
+ * Copyright (c) 2019 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.common.enumeration;
+
+public enum AuthenticationPriority {
```

Comment:
Does this make sense? How do we handle new authentication mechanisms? Like authenticating via GitHub? Would we keep adding time related enums? soonish? before-standard?
Instead of an enum we may just want to use an integer to denote the provider's priority. OR we can have an authentication manager that registers all of the providers and maintains the order instead of each provider setting its own priority.